### ISSUE TITLE
Multiple SubDomainSet with multiple equations per set

### ISSUE THREAD

**Issue by mloubout** [devitocodes/devito#1453]
THe generated code is missing the bound declaration when there is multiple SubDomainSet and multiple equations. The MFE is basically just adding one equation to `tests/test_subdomains.py`::test_multisets:

```
        eq1 = Eq(f, f+2, subdomain=grid.subdomains['mydomains1'])
        eq2 = Eq(g, g+2, subdomain=grid.subdomains['mydomains2'])
        eq3 = Eq(f, f-1, subdomain=grid.subdomains['mydomains1'])

        op = Operator([eq1, eq2, eq3])
```

that produces the following code:

```
#pragma omp parallel num_threads(nthreads)
  {
    #pragma omp for collapse(1) schedule(static,1)
    for (int n = n_m; n <= n_M; n += 1)
    {
      int xi_n_ltkn = xi_n_m[n];
      int yi_n_ltkn = yi_n_m[n];
      int xi_n_rtkn = xi_n_M[n];
      int yi_n_rtkn = yi_n_M[n];
      for (int xi_n = x_m + xi_n_ltkn; xi_n <= x_M - xi_n_rtkn; xi_n += 1)
      {
        #pragma omp simd aligned(f:16)
        for (int yi_n = y_m + yi_n_ltkn; yi_n <= y_M - yi_n_rtkn; yi_n += 1)
        {
          f[xi_n + 1][yi_n + 1] = f[xi_n + 1][yi_n + 1] + 2;
        }
      }
    }
  }
  /* End section0 */
  gettimeofday(&end_section0, NULL);
  timers->section0 += (double)(end_section0.tv_sec-start_section0.tv_sec)+(double)(end_section0.tv_usec-start_section0.tv_usec)/1000000;
  struct timeval start_section1, end_section1;
  gettimeofday(&start_section1, NULL);
  /* Begin section1 */
  #pragma omp parallel num_threads(nthreads)
  {
    #pragma omp for collapse(1) schedule(static,1)
    for (int m = m_m; m <= m_M; m += 1)
    {
      int xi_m_ltkn = xi_m_m[m];
      int yi_m_ltkn = yi_m_m[m];
      int xi_m_rtkn = xi_m_M[m];
      int yi_m_rtkn = yi_m_M[m];
      for (int xi_m = x_m + xi_m_ltkn; xi_m <= x_M - xi_m_rtkn; xi_m += 1)
      {
        #pragma omp simd aligned(g:16)
        for (int yi_m = y_m + yi_m_ltkn; yi_m <= y_M - yi_m_rtkn; yi_m += 1)
        {
          g[xi_m + 1][yi_m + 1] = g[xi_m + 1][yi_m + 1] + 2;
        }
      }
    }
  }
  /* End section1 */
  gettimeofday(&end_section1, NULL);
  timers->section1 += (double)(end_section1.tv_sec-start_section1.tv_sec)+(double)(end_section1.tv_usec-start_section1.tv_usec)/1000000;
  struct timeval start_section2, end_section2;
  gettimeofday(&start_section2, NULL);
  /* Begin section2 */
  #pragma omp parallel num_threads(nthreads)
  {
    #pragma omp for collapse(1) schedule(static,1)
    for (int xi_n = x_m + xi_n_ltkn; xi_n <= x_M - xi_n_rtkn; xi_n += 1)
    {
      #pragma omp simd aligned(f:16)
      for (int yi_n = y_m + yi_n_ltkn; yi_n <= y_M - yi_n_rtkn; yi_n += 1)
      {
        f[xi_n + 1][yi_n + 1] = f[xi_n + 1][yi_n + 1] - 1;
      }
    }
  }

```

and the loop bounds declarations are missing for the last equation. To be investigated (looking into it)

**Comment by rhodrin** [comment]
@mloubout Thanks. Will look into this tomorrow.

**Comment by mloubout** [comment]
@rhodrin 
Line 266 of `devito/operator/operator.py` seems to be the issue

```
if e.subdomain not in seen:
```

which skips the _create_implicit_exprs assuming it will be in the same loop.

**Comment by rhodrin** [comment]
Fixed by #1457.
